src/librustdoc/doctest/make.rs

```diff
@@ -184,6 +184,64 @@ pub(crate) struct DocTestBuilder {
     pub(crate) can_be_merged: bool,
 }
 
+/// Contains needed information for doctest to be correctly generated with expected "wrapping".
+pub(crate) struct WrapperInfo {
+    pub(crate) before: String,
+    pub(crate) after: String,
+    pub(crate) returns_result: bool,
+    insert_indent_space: bool,
+}
+
+impl WrapperInfo {
+    fn len(&self) -> usize {
+        self.before.len() + self.after.len()
+    }
+}
+
+/// Contains a doctest information. Can be converted into code with the `to_string()` method.
+pub(crate) enum DocTestWrapper {
+    Valid { crate_level_code: String, wrapper: Option<WrapperInfo>, code: String },
```

Comment:
1. The name `DocTestWrapper` seems a bit confusing since we already have `DocTest` and `WrapperInfo` (and unlike `WrapperInfo`, this also includes the code itself), maybe this could be `DocTestWrapResult` as it is the result of wrapping the doctest, and it has an error variant like `Result`?
2. couldn't we avoid specifying all the fields twice by leveraging the existing struct? like so: `Valid(DocTest)`

Reply:
> 1. The name `DocTestWrapper` seems a bit confusing since we already have `DocTest` and `WrapperInfo` (and unlike `WrapperInfo`, this also includes the code itself), maybe this could be `DocTestWrapResult` as it is the result of wrapping the doctest, and it has an error variant like `Result`?

Sounds good to me.

> 2\. couldn't we avoid specifying all the fields twice by leveraging the existing struct? like so: `Valid(DocTest)`

You mean `DocTest` from `extracted.rs`? If so I'd rather not. We split internal types with types used for serialization to prevent unexpected changes or to prevent having a custom implementation for the serialization.